doc/source/howtos/encrypt-traffic.rst

```diff
@@ -0,0 +1,307 @@
+
+For more information, see :ref:`state_snapshot_transfer`.
+
+When copying encrypted data via SST,
+the keyring must be sent over with the files for decryption.
+Make sure that the following options are set in :file:`my.cnf` on all nodes:
+
+.. code-block:: text
+
+   early-plugin-load=keyring_file.so
+   keyring-file-data=/path/to/keyring/file
+
+.. warning:: The cluster will not work if keyring configuration across nodes
+   is different.
+
+The following SST methods are available:
+``rsync``, ``mysqldump``, and ``xtrabackup``.
+
+rsync
+-----
+
+This SST method does not support encryption.
+Avoid using this method if you need to secure traffic
+between DONOR and JOINER nodes.
+
+mysqldump
+---------
+
+This SST method dumps data from DONOR and imports it to JOINER.
+Encryption in this case is performed using the SSL certificates configured
+for :ref:`secure MySQL client-server communication <client-server-encryption>`.
+
+Here is how to perform secure SST using ``mysqldump``:
+
+1. Ensure that the DONOR node is configured for SSL encryption
+   (both ``[mysqld]`` and ``[client]`` sections).
+   For more information, see :ref:`client-server-encryption`.
+
+#. Create an SSL user on the DONOR node.
+
+   .. code-block:: sql
+
+      mysql> GRANT USAGE ON *.* TO 'sslsst' REQUIRE SSL;
+
+#. Specify the DONOR superuser credentials
+   in the :variable:`wsrep_sst_auth` variable.
+
+#. Start the JOINER node without Galera library (``--wsrep_provider=none``)
+   and create an SSL user with the same name and grants as on the DONOR node.
+
+#. Configure SSL encryption on JOINER node with the same parameters as DONOR
+   (both ``[mysqld]`` and ``[client]`` sections).
+
+#. Restart JOINER node with Galera library.
+
+If you do everything correctly,
+``mysqldump`` will connect to DONOR using SSL user,
+generate a dump file, and import it to JOINER node.
+
+For more information, see `the relevant section in Galera Cluster documentation <http://galeracluster.com/documentation-webpages/mysqldump.html>`_.
+
+xtrabackup
+----------
+
+This is the default SST method,
+which uses Percona XtraBackup to perform non-blocking transfer of files.
+For more information,
+see :ref:`xtrabackup_sst`.
+
+Encryption mode for this method is selected using the :option:`encrypt` option.
+Depending on the mode you select, other options will be required.
+
+* To enable built-in XtraBackup encryption, use the following configuration:
+
+  .. code-block:: text
+
+     [sst]
+     encrypt=1
+     encrypt-algo=AES256
+     encrypt-key=A1EDC73815467C083B0869508406637E
+
+  In this example, you can set ``encrypt-key-file`` instead of ``encrypt-key``.
+
+  For more information,
+  see `Encrypted Backups <https://www.percona.com/doc/percona-xtrabackup/2.4/innobackupex/encrypted_backups_innobackupex.html>`_.
+
+* To enable SST encryption based on OpenSSL
+  with the certificate authority (``tca``) and certificate (``tcert``) files:
+
+  .. code-block:: text
+
+     [sst]
+     encrypt=2
```

Comment:
1. Again add a deprecated tag
2. Check with @kennt-percona about socat ssl limitation. If socat version doesn't support encryption then we do emit a message.